pkg/security/ImageScanService.go

```diff
@@ -478,37 +477,27 @@ func (impl *ImageScanServiceImpl) ConvertEndStepOutputAndSaveVulnerabilities(ste
 			allSavedCvesMap[cve.Name] = cve
 		}
 	}
+	cvesToUpdate := make([]*repository.CveStore, 0, len(uniqueVulnerabilityMap))
 	for _, vul := range uniqueVulnerabilityMap {
 		var cve *repository.CveStore
 		if val, ok := allSavedCvesMap[vul.Name]; ok {
 			cve = val
+			// updating cve here if vulnerability has a new severity
+			vulnerabilitySeverity := bean.ConvertToSeverityUtility(bean.ConvertToLowerCase(vul.Severity))
+			if vulnerabilitySeverity != cve.Severity {
+				cve.UpdateNewSeverityInCveStore(vulnerabilitySeverity, userId)
+				cvesToUpdate = append(cvesToUpdate, cve)
+			}
 		}
 		if cve == nil {
-			cve = &repository.CveStore{
-				Name:         vul.Name,
-				Package:      vul.Package,
-				Version:      vul.PackageVersion,
-				FixedVersion: vul.FixedInVersion,
-			}
-			lowerCaseSeverity := bean.ConvertToLowerCase(vul.Severity)
-			cve.Severity = bean.ConvertToSeverityUtility(lowerCaseSeverity)
-			cve.StandardSeverity = bean.ConvertToStandardSeverityUtility(lowerCaseSeverity)
-			cve.CreatedOn = time.Now()
-			cve.CreatedBy = userId
-			cve.UpdatedOn = time.Now()
-			cve.UpdatedBy = userId
+			cve = createCveStoreObject(vul.Name, vul.PackageVersion, vul.FixedInVersion, vul.Severity, userId)
```

Comment:
can we bind it to ImageScanOutputObject>

Reply:
I could have bound it to ImageScanOutputObject , but this is being used for  creating  cve obj for clair scanning part also, and their objs types are diff and internal to clair lib we are importing, so there could have been 3 diff bound func, so created one helper func only to serve all three scanning flows